DllImportGenerator/DllImportGenerator/Marshalling/NonBlittableArrayMarshaller.cs

```diff
@@ -65,76 +65,100 @@ public override IEnumerable<StatementSyntax> Generate(TypePositionInfo info, Stu
         {
             var (managedIdentifer, nativeIdentifier) = context.GetIdentifiers(info);
 
+            bool cacheManagedValue = info.IsByRef && context.CanUseAdditionalTemporaryState;
```

Comment:
Maybe pull this out into a helper since `GenerateByteLengthCalculationExpression` and `GenerateNullCheckExpression` use the same logic?